### ISSUE TITLE
Windows package missing liblammps.dll

### ISSUE THREAD

**Issue by martok** [njzjz/lammps-wheel#58]
Hi!

I'm trying to use the LAMMPS wheel on windows, but it appears the wheel does not actually include the `liblammps.dll` library. From the Action logs, it looks like it does get built correctly, but then during the `*** Installing project into wheel...` steps, it does not get packaged.

I have no idea why the import test following the build doesn't fail. Maybe the runner still sees it in some working directory?

This seems to be a known issue in cibuildwheel and the documentation was updated to recommend adding a [repair command](https://cibuildwheel.pypa.io/en/stable/options/#repair-wheel-command) using delvewheel, but does not by default have one. I don't know enough about either tool to recommend anything solid, but it seems like `delvewheel repair ... --include liblammps.dll` could be involved?

Many thanks for having a look!

**Comment by njzjz** [comment]
> `*** Installing project into wheel...`

This message is generated by scikit-build-core but not cibuildwheel.

It looks to me like `lammps.lib` is generated instead of `liblammps.dll`. Maybe related to CMake configurations.

**Comment by njzjz** [comment]
Oh I see that liblammps.dll is built, but it seems that it is not installed by CMake.

>       lammps.vcxproj -> C:\Users\runneradmin\AppData\Local\Temp\tmp9pg6bfjf\build\Release\liblammps.dll

Is there a bug in the LAMMPS CMake file?

**Comment by martok** [comment]
This is where I got the information on it being cibuildwheel's fault from: https://github.com/pypa/cibuildwheel/issues/459

Matplotlib has this in [their config](https://github.com/tacaswell/matplotlib/blob/dceb278bef361f32eb0009dec5bfda41fc27fa38/.github/workflows/cibuildwheel.yml#L97):

```yaml
      CIBW_REPAIR_WHEEL_COMMAND_WINDOWS: >-
        delvewheel repair -w {dest_dir} {wheel}
```

dimod has [something similar](https://github.com/dwavesystems/dimod/pull/1220/files), other than that I can't find much on Github global search. Numpy also uses delvewheel to pack the require libopenblas, but via some wrapper scripts for mingw compatibility reasons.


I'm going to fiddle around with the workflow file and see if I can get this to work. Never worked with Github CI before, so that'll be a learning experience.

**Comment by martok** [comment]
Wait, I think you're right, there is something odd going on with CMake.

From [CMakeFile](https://github.com/lammps/lammps/blob/224468d94640a69197f2506417524a60b071fb8f/cmake/CMakeLists.txt#L826) we get on Linux:

```
  -- Installing: /tmp/tmpr6t6u1n8/wheel/platlib/lammps/./liblammps.so.0
  -- Installing: /tmp/tmpr6t6u1n8/wheel/platlib/lammps/./liblammps.so
  -- Installing: /tmp/tmpr6t6u1n8/wheel/platlib/lammps/./pkgconfig/liblammps.pc
  ...
```

And on Windows, we get the static lib instead of the DLL.

```
    -- Installing: C:\Users\RUNNER~1\AppData\Local\Temp\tmp3t4zvt1n\wheel\platlib\lammps/./lammps.lib
    -- Installing: C:\Users\RUNNER~1\AppData\Local\Temp\tmp3t4zvt1n\wheel\platlib\lammps/./pkgconfig/liblammps.pc
    ...
```

Oh. Yeah, looks like upstream is incorrect here.

> shared libraries - LIBRARY
> 
>     Created by add_library with the SHARED option on UNIX-like platforms (.so on most UNIX, .dylib on Mac)
> dynamic-link libraries - RUNTIME
> 
>     Created by add_library with the SHARED option on Windows platforms (.dll)

So the line linked above should at least include `RUNTIME DESTINATION ...` for Windows.

Nothing to do here then, I'll open a PR upstream once I'm sure that doesn't break the other platforms.

**Comment by martok** [comment]
Done and merged upstream: lammps/lammps#4353

Hopefully this will make it to the next update for the stable release branch.
